js/modules/k6/k6.go

```diff
@@ -91,6 +92,9 @@ func (*K6) Group(ctx context.Context, name string, fn goja.Callable) (goja.Value
 	t := time.Now()
 
 	tags := state.Options.RunTags.CloneTags()
+	for k, v := range state.Tags {
+		tags[k] = v
+	}
```

Comment:
Hm ... shouldn't it just be `state.Tags.CloneTags()` ? And have the `state.Options.RunTags` be cloned to `state.Tags` along all the others ?

Reply:
Not sure I'm following, sorry. Can you elaborate on that?

`Options.RunTags` are the globally defined tags, whereas `state.Tags` was added here to pass on the custom executor tags. There's an overriding hierarchy here, which we might want to document.

Reply:
I am just of the opinion you can put the overridden final tags inside `state.Tags` and use them everywhere where `state.Options.RunTags` are being used currently instead of merging them everywhere

Reply:
Aha, I see. I'll look into that, thanks.

Reply:
@MStoykov See d15bfe93 for this change.

I'm a bit apprehensive about it, since it could break some previous overriding, and it couldn't be done in [`httpext.MakeRequest`](https://github.com/loadimpact/k6/blob/d15bfe93177fabfd80527045a270930b2f609eed/lib/netext/httpext/request.go#L234), otherwise `TestRequestAndBatch/Params/tags/tags-precedence` would fail. Also now WS metrics will be tagged with optional system tags (`iter`, `vu`, etc.), whereas previously they weren't. Arguably an improvement, but a bit worrying that something else didn't inadvertently change and we're not testing for it.

I'd leave this in anyway, but there's now a race condition in CI I thought was fixed by `e33400e5`, which passed locally, but apparently not, or I broke it again. :-/

Reply:
I don't know ... I think this is once again too big of change ... and at this point, I think it was a mistake to try and implement both exec, tags, and env at the same time. Especially tags seem to be problematic, mostly because they were problematic even before that. I don't know what is the correct change, but am fine with you reverting the changes ... or not - I am of the opinion that this probably fixes some tiems when `vu`/`iter` weren't present  in metrics they should've been in. 

The race conditions on the other hand are problematic and will need to be fixed :(

Reply:
cc @na--

Reply:
The race condition seems to be between https://github.com/loadimpact/k6/blob/d15bfe93177fabfd80527045a270930b2f609eed/lib/executor/variable_looping_vus.go#L579 and https://github.com/loadimpact/k6/blob/d15bfe93177fabfd80527045a270930b2f609eed/lib/executor/variable_looping_vus.go#L587, where apparently by https://golang.org/src/sync/waitgroup.go?s=3500:3527#L124 you need to synchronize the Add and Wait ... which I agree with and it is something that my WIP change does for variable arrival rate ... I think ... 

So *maybe* this isn't a problem with this PR :D

Reply:
Yeah, while it doesn't seem related to this PR, the raciness was exhibited with the new tests, which is good I suppose, since I didn't find anything wrong with the tests themselves.

So I applied the same `defer` reordering fix we in a previous PR for the arrival-rate execs to the VLV one, and it seemed to fix the issue locally at least (although that could be flaky...). See 60b2ffe9. Let's wait for CI...